dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/DataProvider.java

```diff
@@ -896,33 +895,21 @@ public void moduleStarting(ModuleInfo moduleInfo) throws StateChangeException {
         Set<FutureEMBuilder> futures = futureEMBuilders.get(appName);
         if (futures != null)
             for (FutureEMBuilder futureEMBuilder : futures) {
-                // The JEE name includes .jar at the end of EJB modules
-                String moduleNameWithDot = futureEMBuilder.jeeName.getModule();
-
-                if (!futureEMBuilder.inWebModule &&
-                    moduleNameWithDot != null &&
-                    moduleNameWithDot.length() == moduleName.length() + 4 &&
-                    moduleNameWithDot.startsWith(moduleName) &&
-                    moduleNameWithDot.endsWith(".jar")) {
-
-                    if (trace && tc.isDebugEnabled())
-                        Tr.debug(this, tc, "matched with " + futureEMBuilder.jeeName);
-
-                    // This delays createEMBuilder until restore.
-                    // While this works by avoiding all connections to the data source, it does make restore much slower.
-                    // TODO figure out how to do more work on restore without having to make a connection to the data source
-                    CheckpointPhase.onRestore(() -> futureEMBuilder.completeAsync(futureEMBuilder::createEMBuilder, executor));
+                // This delays createEMBuilder until restore.
+                // While this works by avoiding all connections to the data source, it does make restore much slower.
+                // TODO figure out how to do more work on restore without having to make a connection to the data source
+                CheckpointPhase.onRestore(() -> futureEMBuilder.completeAsync(futureEMBuilder::createEMBuilder, executor));
+
+                if (processed == null) {
+                    processed = new ConcurrentSkipListSet<>();
+                    Set<FutureEMBuilder> previous = futureEMBuildersInEJB //
+                                    .putIfAbsent(appName, processed);
```

Comment:
Given that this was moved out from under the check to detect EJB modules, does this mean that we are now adding all of the FutureEMBuilders for web modules into futureEMBuildersInEJB ?  If so, this is confusing.